### PR TITLE
boards: riscv: longan_nano: add adc support

### PR DESCRIPTION
**Pull request by maxinbjohn**
Add ADC support for longan nano.

Tested using samples/drivers/adc

### CODE REVIEW

**Review by cameled — COMMENTED** [review]
@maxinbjohn  Thanks for you contribution! 

Please add `ADC`  support to longan_nano document `Supported Features` section, which is under `zephyr/boards/riscv/longan_nano/doc/index.rst`.

Also,  please consider to add longgan_nano dts overlay to `zephyr/samples/drivers/adc/boards`, which you can refer `gd32f450i_eval.overlay` implement.

After these is done, new user can directly to run `adc` sample for langan_nano board.

File: boards/riscv/longan_nano/longan_nano_lite.yaml [hunk 1]
```diff
@@ -8,6 +8,7 @@ toolchain:
 flash: 64
 ram: 20
 supported:
+  - adc
```
Comment (cameled) [inline boards/riscv/longan_nano/longan_nano_lite.yaml:11]: I think this yaml file is used for `tests`, and there have a CI failure for `adc_api` test.

Please remove `adc` support from this file or add related dts overlay to `zephyr/tests/drivers/adc/adc_api/boards`.

File: boards/riscv/longan_nano/longan_nano.yaml [hunk 2]
```diff
@@ -8,6 +8,7 @@ toolchain:
 flash: 128
 ram: 32
 supported:
+  - adc
```
Comment (cameled) [inline boards/riscv/longan_nano/longan_nano.yaml:11]: +1

File: boards/riscv/longan_nano/longan_nano_lite.yaml (hunk 1 above)
Comment (maxinbjohn) [reply boards/riscv/longan_nano/longan_nano_lite.yaml:11]: Thank you for looking into that. I will update it as per your comments.

File: samples/drivers/adc/sample.yaml [hunk 3]
```diff
@@ -26,6 +26,8 @@ tests:
       - gd32l233r_eval
       - lpcxpresso55s36
       - mr_canhubk3
+      - longan_nano
+      - longan_nano_lite
```
Comment (cameled) [inline samples/drivers/adc/sample.yaml:30]: Does longan_nano_lite is different board than langan_nano? If so we need lite dts overlay too.

Comment (maxinbjohn) [reply samples/drivers/adc/sample.yaml:30]: I will update it. Thanks!

**Review by cameled — APPROVED** [review]
LGTM.

**Comment by cameled** [comment]
@nandojve please have a look.

**Review by nandojve — APPROVED** [review]

**Review by soburi — APPROVED** [review]

**Review by anangl — APPROVED** [review]

**Pull request merged**